### PR TITLE
threadstore: make apis return errors

### PR DESCRIPTION
**Pull request by jsign**
This is an API change to consider errors in most Books APIs, since datastore backed books can have errors at most kind of actions.

This PR is a pre-requisite to possibly merge a pending  [`go-textile-threads` PR](https://github.com/textileio/go-textile-threads/pull/21).

I'll change from Draft when [`go-textile-threads` #15](https://github.com/textileio/go-textile-threads/pull/15)  gets merged.

### CODE REVIEW

**Comment by jsign** [comment]
I think this should be ready to review. While fixing `go-textile-threads`, the returned errors bubbled up to other APIs (last commit of this PR).

After merged I think we could tag it, so to start having better version references in other `go.mod`s. I think living in the `v.0.0.x` for a while makes sense until it stabilizes. Since this is mostly a repo about APIs, I imagine most version changes will be major ones. The only minor or patch things would be new _utils_ or their fixes (crypto things, etc). What do you think @sanderpick ?

**Review by sanderpick — APPROVED** [review]
LGTM

**Comment by sanderpick** [comment]
Regarding versioning - perhaps we can hold off until https://github.com/textileio/go-textile-core/pull/14 goes in - same for threads, hold off until https://github.com/textileio/go-textile-threads/pull/22 goes in

**Comment by jsign** [comment]
> Regarding versioning - perhaps we can hold off until #14 goes in - same for threads, hold off until [textileio/go-textile-threads#22](https://github.com/textileio/go-textile-threads/pull/22) goes in
 :+1:

**Pull request merged**
